### PR TITLE
[Simulator] Add SWAN client interface

### PR DESCRIPTION
**Pull request by IvanPombo**
TLDR: Adds the SWAN simulator to the client.

## Description

In this PR, we add the client interface to execute simulations with SWAN simulator that was added to the backend in the PR inductiva/inductiva-web-api#632.

This simulator runs exactly as SWASH with a single binary and MPI, so the user only needs to pass the input directory and the name of `sim_config_filename`. An example input dir was added to our respective cloud folder to allow for testing.

## Implications

The main application, that should be addressed soon and that here I was not strict is with the `api_method_name`. In a next PR, I will make sure to uniformize all of them together with the backend.

There are few threads open with SWAN, respectively the usage of OpenMP vs MPI when running the simulator. One discussion in the "literature" is that OpenMP is more effective in a single machine, while MPI is more effective over several. Here, we would need to separate images for each, since SWAN doesn't seem to allow for compilation of both at the same time. Due to our current structure and simplicity, I have followed with the implementation with MPI, and further conversations with users may decide our next steps.

A README will follow on a next iteration.

## Test

To test this simulator you can deploy the backend locally or wait for it to be merged into development. To do so, use the following script:

```python
import inductiva

# Set simulation input directory
input_dir = inductiva.utils.files.download_from_url(
    "https://storage.googleapis.com/inductiva-api-demo-files/"
    "swan-input-example.zip", True)

# Initialize the Simulator
swan = inductiva.simulators.SWAN()

# Run simulation with config files in the input directory
task = swash.run(
    input_dir=input_dir, sim_config_filename="a11refr.swn")
```

### CODE REVIEW

**Review by Pbarbosa92 — APPROVED** [review]

**Review by AugustoPeres — APPROVED** [review]

**Review by sergiomsantos — COMMENTED** [review]
We can also add the associated doc page
